apps/prairielearn/src/server.ts

```diff
@@ -2035,7 +2042,14 @@ function idleErrorHandler(err: Error) {
   Sentry.close().finally(() => process.exit(1));
 }
 
-if (esMain(import.meta) && config.startServer) {
+if (!((esMain(import.meta) || import.meta.env?.DEV) && config.startServer)) {
```

Comment:
I don't fully understand the purpose of this guard. All code below it is in the guard, so it would make more sense to exit early.